### PR TITLE
Adding IAM resources for Dataplex Lake, Zone and Asset

### PR DESCRIPTION
**Pull request by dbjnbnrj**
Adding get/setIAMPolicy support for Dataplex Lake, Zone and Asset
Lake, Zone and Asset resources were generated via DCL. Followed guide https://googlecloudplatform.github.io/magic-modules/docs/how-to/add-mmv1-iam/#adding-iam-support-to-nonexistent-resources to add the IAM resources.
(part of https://github.com/hashicorp/terraform-provider-google/issues/11648)

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_dataplex_lake_iam_*`
```
```release-note:new-resource
`google_dataplex_zone_iam_*`
```
```release-note:new-resource
`google_dataplex_asset_iam_*`
```

### CODE REVIEW

**Comment by shuyama1** [comment]
@dbjnbnrj Thanks for making the PR. Would you mind rebasing your PR to kick in a recent CI change? Thanks

**Comment by dbjnbnrj** [comment]
Hi @shuyama1, 

Can you help me debug the VCRTest run? I'm unusure what the root cause is and my local runs have been successful - 

```
        Error: Error setting IAM policy for dataplex zone "projects/ci-test-project-188019/locations/us-central1/lakes/tf-test-lakeesn7g39rdj/zones/tf-test-zoneesn7g39rdj": Post "projects/ci-test-project-188019/locations/us-central1/lakes/tf-test-lakeesn7g39rdj/zones/tf-test-zoneesn7g39rdj:setIamPolicy?alt=json": unsupported protocol scheme ""
```

**Review by shuyama1 — COMMENTED** [review]
Would you mind splitting the `api.yaml file` into individual file for each resource. Sorry we recently did the file split and now product level file `api.yaml` should no longer be added

File: tpgtools/overrides/dataplex/tpgtools_product.yaml
Comment (dbjnbnrj) [reply tpgtools/overrides/dataplex/tpgtools_product.yaml:1]: Thanks for the suggestion. Updated this.

**Comment by dbjnbnrj** [comment]
Hi @shuyama1  looks like the two tests that failed are unrelated to my change
TestAccComposerEnvironment_withEncryptionConfigComposer2 
TestAccDataSourceDnsManagedZone_basic

**Comment by shuyama1** [comment]
> Hi @shuyama1 looks like the two tests that failed are unrelated to my change TestAccComposerEnvironment_withEncryptionConfigComposer2 TestAccDataSourceDnsManagedZone_basic

Those failures are unrelated. Don't worry about them! Sorry for the noise.

**Comment by dbjnbnrj** [comment]
Hi @shuyama1 looks like there was one failure, merged from main. Let me know if you can help! Thanks :)

**Comment by dbjnbnrj** [comment]
friendly ping @shuyama1

**Comment by shuyama1** [comment]
Thank you for the ping! Overall LGTM with only a restructure work needed! Sorry that looks like we have another provider rework PR that removed the terraform.yaml got merged recently. Would you mind copying the `iam_policy` code block from terraform.yaml to each resource.yaml file. Really apologize for the extra work. I think the PR should be good to go after. Also don't worry about the unrelated test failure. That's a known flaky test and should not block the merge of this PR.

**Comment by dbjnbnrj** [comment]
> Thank you for the ping! Overall LGTM with only a restructure work needed! Sorry that looks like we have another provider rework PR that removed the terraform.yaml got merged recently. Would you mind copying the `iam_policy` code block from terraform.yaml to each resource.yaml file. Really apologize for the extra work. I think the PR should be good to go after. Also don't worry about the unrelated test failure. That's a known flaky test and should not block the merge of this PR.

Sure no worries, should I also include the iam_policy in the terraform.yaml file as well? Thanks!

**Comment by shuyama1** [comment]
> > Thank you for the ping! Overall LGTM with only a restructure work needed! Sorry that looks like we have another provider rework PR that removed the terraform.yaml got merged recently. Would you mind copying the `iam_policy` code block from terraform.yaml to each resource.yaml file. Really apologize for the extra work. I think the PR should be good to go after. Also don't worry about the unrelated test failure. That's a known flaky test and should not block the merge of this PR.
> 
> Sure no worries, should I also include the iam_policy in the terraform.yaml file as well? Thanks!

No, the terraform.yaml file should be removed. Sorry, I should've used the word `move` instead of `copy`. Sorry for the confusion

**Comment by dbjnbnrj** [comment]
Thanks, removed terraform.yaml

**Comment by dbjnbnrj** [comment]
Hi @shuyama1 looks like workflow is waiting for approval from maintainer. Let me know if you can help

**Comment by shuyama1** [comment]
tests failed due to:
```
google-beta/iam_dataplex_asset.go:48:2: duplicate key "lake" in map literal
google-beta/iam_dataplex_asset.go:61:2: lake redeclared
	google-beta/iam_dataplex_asset.go:59:2: other declaration of lake
google-beta/iam_dataplex_asset.go:110:7: duplicate field name lake in struct literal
google-beta/iam_dataplex_asset.go:161:7: duplicate field name lake in struct literal
google-beta/iam_dataplex_zone.go:43:2: duplicate key "lake" in map literal
google-beta/iam_dataplex_zone.go:55:2: lake redeclared
	google-beta/iam_dataplex_zone.go:54:2: other declaration of lake
google-beta/iam_dataplex_zone.go:99:7: duplicate field name lake in struct literal
google-beta/iam_dataplex_zone.go:146:7: duplicate field name lake in struct literal
make: *** [vet] Error 2
GNUmakefile:26: recipe for target 'vet' failed
make: *** No rule to make target 'tools'.  Stop.
==> Checking that code complies with gofmt requirements...
go vet
# github.com/hashicorp/terraform-provider-google-beta/google-beta
google-beta/iam_dataplex_asset.go:48:2: duplicate key "lake" in map literal
google-beta/iam_dataplex_asset.go:61:2: lake redeclared
	google-beta/iam_dataplex_asset.go:59:2: other declaration of lake
google-beta/iam_dataplex_asset.go:110:7: duplicate field name lake in struct literal
google-beta/iam_dataplex_asset.go:161:7: duplicate field name lake in struct literal
google-beta/iam_dataplex_zone.go:43:2: duplicate key "lake" in map literal
google-beta/iam_dataplex_zone.go:55:2: lake redeclared
	google-beta/iam_dataplex_zone.go:54:2: other declaration of lake
google-beta/iam_dataplex_zone.go:99:7: duplicate field name lake in struct literal
google-beta/iam_dataplex_zone.go:146:7: duplicate field name lake in struct literal
make: *** [vet] Error 2
GNUmakefile:26: recipe for target 'vet' failed
==> Checking that code complies with gofmt requirements...
go vet
# github.com/hashicorp/terraform-provider-google-beta/google-beta
google-beta/iam_dataplex_asset.go:48:2: duplicate key "lake" in map literal
google-beta/iam_dataplex_asset.go:61:2: lake redeclared
	google-beta/iam_dataplex_asset.go:59:2: other declaration of lake
google-beta/iam_dataplex_asset.go:110:7: duplicate field name lake in struct literal
google-beta/iam_dataplex_asset.go:161:7: duplicate field name lake in struct literal
google-beta/iam_dataplex_zone.go:43:2: duplicate key "lake" in map literal
google-beta/iam_dataplex_zone.go:55:2: lake redeclared
	google-beta/iam_dataplex_zone.go:54:2: other declaration of lake
google-beta/iam_dataplex_zone.go:99:7: duplicate field name lake in struct literal
google-beta/iam_dataplex_zone.go:146:7: duplicate field name lake in struct literal
GNUmakefile:26: recipe for target 'vet' failed
make: *** [vet] Error 2
```

Looking into the issue

File: mmv1/products/dataplex/Zone.yaml
Comment (shuyama1) [inline mmv1/products/dataplex/Zone.yaml]: Also here

**Review by shuyama1 — CHANGES_REQUESTED** [review]

**Comment by shuyama1** [comment]
Looks like you've already resolved it. Thanks for the quick fix.
/gcbrun

**Comment by dbjnbnrj** [comment]
Unsure why this is still failing, think the issue should be resolved now. Is the diff by any chance picking up the previous version?

**Comment by shuyama1** [comment]
Looks like the provider tests passed which means the issue should be resolved, but the validator tests failed. Let me rerun the tests and also I'll grab someone who knows the validator better to take a look.
/gcbrun

**Comment by shuyama1** [comment]
Looks like it got past the previous failure point.

Thanks for taking a look @melinath!

**Review by shuyama1 — APPROVED** [review]
Thank you!

**Comment by dbjnbnrj** [comment]
Thank you so much for your help @shuyama1!! Thanks @melinath for helping with the failures :D

**Pull request merged**
